From 82d1b7ecdaac5a7f852a6106cefca67bc14e4792 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Tue, 12 Feb 2019 19:00:28 +0100
Subject: Wrap several native binary calls on Yocto building with qemu.

The patch below makes the V8 binaries run during the build be invoked through
QEMU, as they are built for the target:
* mksnapshot
* torque
* gen-regexp-special-case

Based on original patch from Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>

Upstream-Status: Inappropriate [embedded specific]

diff --git a/BUILD.gn b/BUILD.gn
index f9d080c908..14e32768fa 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1036,6 +1036,7 @@ action("run_torque") {
   }
 
   args = [
+    "./v8-qemu-wrapper.sh",
     "./" + rebase_path(get_label_info(":torque($v8_generator_toolchain)",
                                       "root_out_dir") + "/torque",
                        root_build_dir),
@@ -1114,6 +1115,7 @@ action("generate_bytecode_builtins_list") {
     ":bytecode_builtins_list_generator($v8_generator_toolchain)",
   ]
   args = [
+    "./v8-qemu-wrapper.sh",
     "./" + rebase_path(
             get_label_info(
                     ":bytecode_builtins_list_generator($v8_generator_toolchain)",
@@ -1157,6 +1159,7 @@ template("run_mksnapshot") {
     data = []
 
     args = [
+      "./v8-qemu-wrapper.sh",
       "./" + rebase_path(get_label_info(":mksnapshot($v8_snapshot_toolchain)",
                                         "root_out_dir") + "/mksnapshot",
                          root_build_dir),
@@ -3926,6 +3929,7 @@ if (v8_enable_i18n_support) {
     ]
 
     args = [
+      "./v8-qemu-wrapper.sh",
       "./" + rebase_path(
               get_label_info(
                       ":gen-regexp-special-case($v8_generator_toolchain)",
-- 
2.20.1

